hudi-utilities/src/test/java/org/apache/hudi/utilities/TestHoodieDeltaStreamer.java

```diff
@@ -428,7 +446,11 @@ public void testBulkInsertsAndUpsertsWithSQLBasedTransformerFor2StepPipeline() t
     TestHelpers.assertCommitMetadata("00000", datasetBasePath, dfs, 1);
 
     // with no change in upstream table, no change in downstream too when pulled.
-    new HoodieDeltaStreamer(downstreamCfg, jsc).sync();
+    try {
+      new HoodieDeltaStreamer(downstreamCfg, jsc).sync();
+    } catch (Exception e) {
+      log.error("expected error due to null schema provider!");
+    }
```

Comment:
I think catch the exception here breaks the original intention of the 2 step pipeline, it should run normally with no new change pulled, also just catch exception here achieve the same goal that the exception will be thrown when schema provider is null with the following UT(testNullSchemaProvider). 
I write the TestHoodieDeltaStreamer(mainly define a `DummySchemaProvider class ` to make the pipeline works normally and simplify the `makeConfigWithoutSchemaProvider` with `makeConfig` methods) in the [gist](https://gist.github.com/leesf/3da5439c242d0decae4b324838eb7f5d) which you would refer to. WDYT? @pratyakshsharma CC @vinothchandar

Reply:
Regarding simplifying makeConfigWithoutSchemaProvider method, initially I was thinking in a similar way like you proposed. I get your point (I should always try to write minimal code).
I will make both the changes. @leesf